### DISCUSSION TITLE
export HFSS model history

### DISCUSSION THREAD

**Discussion by erikajob91** [ansys/pyaedt#5434] [Q&A] [answered]
I'd like to know if there is any way to export the whole HFSS model history in order to get a script for rebuilding the same model in an automatic way.

**Comment by gmalinve** [comment, marked as answer]
Hi @erikajob91,

Thanks for opening this discussion.
Right now we don't have any feature in PyAEDT that allows you to export the history tree of your model.
What we do have is the method `export_config` that exports the current design properties to a JSON or TOML file.
This method export all your model's information, including objects info (without history).

You can export the history of an object as a json like:

```
import json
with open(output_file, "w") as outfile:
     json.dump(your_object.jsonalize_tree(), output_file)
```

but right now PyAEDT `import_config `doesn't read this json.

Hope this helps, 

Kind regards,

Giulia
